config/schema/mongodb-1.0.xsd

```diff
@@ -119,6 +123,82 @@
     <xsd:attribute name="value" type="xsd:string" use="required" />
   </xsd:complexType>
 
+  <xsd:complexType name="auto-encryption">
+    <xsd:sequence>
+      <xsd:element name="kmsProvider" type="kms-provider" minOccurs="0" maxOccurs="1" />
+      <xsd:element name="masterKey" type="master-key" minOccurs="0" maxOccurs="1" />
+      <xsd:element name="keyVaultNamespace" type="xsd:string" minOccurs="0" maxOccurs="1" />
+      <xsd:element name="tlsOptions" type="tls-options" minOccurs="0" maxOccurs="1" />
+      <xsd:element name="encryptedFieldsMap" type="encrypted-fields-map" minOccurs="0" maxOccurs="1" />
+      <xsd:element name="extraOptions" type="extra-options" minOccurs="0" maxOccurs="1" />
+    </xsd:sequence>
+    <xsd:attribute name="bypassAutoEncryption" type="xsd:boolean" use="optional" />
+    <xsd:attribute name="bypassQueryAnalysis" type="xsd:boolean" use="optional" />
+  </xsd:complexType>
+
+  <xsd:complexType name="kms-provider">
+    <xsd:attribute name="type" type="xsd:string" use="required" />
+    <!-- AWS -->
+    <xsd:attribute name="accessKeyId" type="xsd:string" use="optional" />
+    <xsd:attribute name="secretAccessKey" type="xsd:string" use="optional" />
+    <xsd:attribute name="sessionToken" type="xsd:string" use="optional" />
+    <!-- Azure -->
+    <xsd:attribute name="tenantId" type="xsd:string" use="optional" />
+    <xsd:attribute name="clientId" type="xsd:string" use="optional" />
+    <xsd:attribute name="clientSecret" type="xsd:string" use="optional" />
+    <xsd:attribute name="keyVaultEndpoint" type="xsd:string" use="optional" />
+    <xsd:attribute name="identityPlatformEndpoint" type="xsd:string" use="optional" />
+    <xsd:attribute name="keyName" type="xsd:string" use="optional" />
+    <xsd:attribute name="keyVersion" type="xsd:string" use="optional" />
+    <!-- GCP -->
+    <xsd:attribute name="email" type="xsd:string" use="optional" />
+    <xsd:attribute name="privateKey" type="xsd:string" use="optional" />
+    <xsd:attribute name="endpoint" type="xsd:string" use="optional" />
+    <xsd:attribute name="projectId" type="xsd:string" use="optional" />
+    <xsd:attribute name="location" type="xsd:string" use="optional" />
+    <xsd:attribute name="keyRing" type="xsd:string" use="optional" />
+    <!-- <xsd:attribute name="keyName" type="xsd:string" use="optional" /> -->
+    <!-- <xsd:attribute name="keyVersion" type="xsd:string" use="optional" /> -->
+    <!-- KMIP -->
+    <!-- <xsd:attribute name="endpoint" type="xsd:string" use="optional" /> -->
+    <xsd:attribute name="tlsCAFile" type="xsd:string" use="optional" />
+    <xsd:attribute name="tlsClientCertificateKeyFile" type="xsd:string" use="optional" />
+    <xsd:attribute name="tlsClientCertificateKeyFilePassword" type="xsd:string" use="optional" />
```

Comment:
Related to my comment on the configuration, I don't think any KMS provider has these TLS options.